src/main/java/com/puppycrawl/tools/checkstyle/utils/AnnotationUtil.java

```diff
@@ -167,4 +169,31 @@ public static DetailAST getAnnotation(final DetailAST ast,
         return result;
     }
 
+    /**
+     * Checks if the given AST element is marked with an annotation that allows
+     * it to omit javadoc documentation.
```

Comment:
This is a utility method, so there should be no connection to javadocs.

Reply:
Good point, I will rephrase the comment and rename the method

Reply:
I did some refactoring as well. The build will fail due to code coverage but I will deal with that after you've had a chance to see the changes.

Reply:
fixed